sklearn/metrics/tests/test_ranking.py

```diff
@@ -737,8 +737,9 @@ def _test_precision_recall_curve(y_true, probas_pred):
     assert_array_almost_equal(precision_recall_auc, 0.859, 3)
     assert_array_almost_equal(precision_recall_auc,
                               average_precision_score(y_true, probas_pred))
+    # `_average_precision` is not very precise in case of 0.5 ties: be tolerant
     assert_almost_equal(_average_precision(y_true, probas_pred),
-                        precision_recall_auc, decimal=3)
+                        precision_recall_auc, decimal=2)
```

Comment:
is this related?

Reply:
If I remember well, it is. Changing the random number generation seems to make this test fail. When I investigated a few months ago, I could not find a good reason except for precision of the `_average_precision` in case of 0.5 ties.

I would certainly appreciate a second pass of checking on that one, to confirm my impression.